proposals/proposal_1/lib/src/logging/logging_service.dart

```diff
@@ -0,0 +1,25 @@
+/// A Service that records application messages or exceptions.
```

Comment:
I'm happy to explore that alternative, but can you please describe your use cases in a bit more detail? Or perhaps demonstrate with code if that's easier than an explanation?

From my experience, you have a few use cases for logging:

  - Print logs in Debug
  - Report information to a service in release mode. 
  - Disable printing in profile mode, since it adds noise to the benchmark results. Disable reporting to a service in profile mode, since profiles may run many many times on CI servers and you do not want to send junk data to your logging service.

While package:logging  gives you something send logs to, it does not provide a structure to solve these particular use cases. Rather, it's a bit like a Logging Event Bus system. To perform the tasks I mentioned, you'd still need to do something like this:

```dart
Logger.root.onRecord.listen((record) {
 // perform all of the logic here to send the record to the correct backend depending on the environment.
  if (kDebugMode) {
    print('${record.level.name}: ${record.time}: ${record.message}');
  } else if (kReleaseMode) {
    switch (record.level) {
      case Level.Severe:
        // send it to sentry
    }
  }
});

// or Still use the service depending on the environment...
final loggingService = kDebugMode ? DebugLoggingService()...;

Logger.root.onRecord.listen((record) {
  switch (record.level) {
      case Level.Severe:
        _loggingService.error(...);
    }
});
```

IMO, this has a few drawbacks:

  1. You couple your application to the Logger class directly, instead of hiding the Logger behind a class you control. This becomes a problem if you need add more information to your logs.
  2. You still need to *do* something with those records, and doing something becomes more complex -- generally nested switch or if statements. 
  3. Difficult to trace the execution path. With `loggingService.info` you can jump-to-definition to see what it's doing. You cannot do the same with `Logger().info` -- you need to find the correct `onRecord.listen` and trace through the nested ifs/switches.

Therefore, I've always found it easier to leave the logging package out of the apps I work on. What do you think?

Reply:
@felangel Actually I'd be curious to hear how you all handle logging. Do you make use of the Logging package? If so, how do you structure it?

Reply:
@brianegan it's funny you should ask. In the past we've made use of the [lumberdash](https://pub.dev/packages/lumberdash) logging client but for various reasons we've decided to move away from it and are in the process of evaluating alternative approaches 😄 

With lumberdash, we'd initialize different loggers depending on if the app was running in debug or release mode. For example, in release mode we might use [firebase_lumberdash](https://pub.dev/packages/firebase_lumberdash) or [sentry_lumberdash](https://pub.dev/packages/sentry_lumberdash) whereas in debug mode we might just use [print_lumberdash](https://pub.dev/packages/print_lumberdash).

fwiw I've been playing around with using the [built-in log api](https://api.flutter.dev/flutter/dart-developer/log.html) to see if that could be extended to support these various use-cases.

Reply:
Ah, Lumberdash is a very nice package as well :) Good to hear we're thinking of similar use-cases. 

Thanks again for your help -- really appreciate your time and feedback!